src/site/content/en/blog/css-module-scripts/index.md

```diff
@@ -0,0 +1,137 @@
+  modules.
+description: >
+  Learn how to use CSS module scripts to import CSS stylesheets using the same syntax as JavaScript
+  modules.
+authors:
+  - dandclark
+date: 2021-08-06 # TODO Update to publishing date
+tags:
+  - blog
+  - css
+  - DOM
+  - javascript
+  - modules
+---
+
+With the new CSS module scripts feature, you can load CSS style sheets with `import` statements,
+just like [JavaScript
+modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules). The style sheets
+can then be applied to documents or [shadow
+roots](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot) in the same manner as
+[constructable
+stylesheets](https://developers.google.com/web/updates/2019/02/constructable-stylesheets). This can
+be more convenient and [more
+performant](https://dandclark.github.io/json-css-module-notes/#css-module-performancememory-examples)
+than other ways of importing and applying CSS.
+
+## Browser Support
+
+CSS module scripts are available in Chrome and Edge in version 93. 
+
+Support in Firefox and Safari is not yet available. Implementation progress can be tracked at the
+[Gecko bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1720570) and [WebKit
+bug](https://bugs.webkit.org/show_bug.cgi?id=227967), respectively.
+
+## Prerequisites
+
+- Familiarity with [JavaScript
+  modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules).
+- Familiarity with [constructable
+  stylesheets](https://developers.google.com/web/updates/2019/02/constructable-stylesheets).
+
+## Using CSS module scripts
+
+Import a CSS module script and apply it to a document or a shadow root like this:
+
+```js
+import sheet from './styles.css' assert { type: 'css' };
+document.adoptedStyleSheets = [sheet];
+shadowRoot.adoptedStyleSheets = [sheet];
+```
+
+The default export of a CSS module script is a [constructable
+stylesheet](https://developers.google.com/web/updates/2019/02/constructable-stylesheets) whose
+contents are those of the imported file. Like any other constructable stylesheet, it is applied to
+documents or shadow roots using
+[`adoptedStyleSheets`](https://wicg.github.io/construct-stylesheets/#using-constructed-stylesheets).
+
+Unlike other ways of applying CSS from JavaScript, there is no need to create `<style>` elements or
+mess with JavaScript strings of CSS text.
+
+CSS modules also have some of the same benefits as JavaScript modules.
+- Deduplication: if the same CSS file is imported from multiple places in an application, it will
+  still only be fetched, instantiated, and parsed a single time.
+- Consistent order of evaluation: when the importing JavaScript is running, it can rely on the
+  stylesheet it imports having already been fetched and parsed.
+- Security: modules are fetched with [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+  and use strict MIME-type checking.
```

Comment:
```suggestion
CSS modules also have some of the same benefits as JavaScript modules.
- **Deduplication:** if the same CSS file is imported from multiple places in an application, it will
  still only be fetched, instantiated, and parsed a single time.
- **Consistent order of evaluation:** when the importing JavaScript is running, it can rely on the
  stylesheet it imports having already been fetched and parsed.
- **Security:** modules are fetched with [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
  and use strict MIME-type checking.
```